packages/rollup-plugin-html/src/output/adjustToFullAbsoluteUrls.ts

```diff
@@ -0,0 +1,47 @@
+import { Node, Document } from 'parse5';
+import { findElements, getTagName, getAttribute, setAttribute } from '@web/parse5-utils';
+
+function isAbsoluteableNode(node: Node) {
+  const metaAttributes = ['og:url', 'og:image'];
+  switch (getTagName(node)) {
+    case 'link':
+      if (getAttribute(node, 'rel') === 'canonical' && getAttribute(node, 'href')) {
+        return true;
+      }
+      return false;
+    case 'meta':
+      if (
+        metaAttributes.includes(getAttribute(node, 'property')!) &&
+        getAttribute(node, 'content')
+      ) {
+        return true;
+      }
+      return false;
+    default:
+      return false;
+  }
+}
+
+function joinSiteUrlAndPart(siteUrl: string, part: string) {
+  if (part.startsWith('http')) {
+    return part;
+  }
+  if (part === '/') {
+    return siteUrl + part;
+  }
+  return `${siteUrl}/${part}`;
+}
+
+export function adjustToFullAbsoluteUrls(document: Document, siteUrl: string) {
```

Comment:
```suggestion
export function injectAbsoluteUrls(document: Document, siteUrl: string) {
```

Keeps it consistent with the other functions. Perhaps it should be called from `injectedUpdatedAssetPaths` as well.

Reply:
thought about it - injecting means to me to add new nodes... that's why I used a different name... but I can change it to whatever

Reply:
You're injecting different paths. But we could call it rewriting as well. At least both should use the same naming.

Reply:
used `injectAbsoluteBaseUrl`